logos-codegen/src/parser/mod.rs

```
@@ -298,6 +353,23 @@ impl Parser {
         Some(inline.into())
     }
 
+    fn parse_skip_callback(&mut self, tokens: TokenStream) -> Option<SkipCallback> {
+        let span = tokens.span();
+        Some(match self.parse_callback(tokens) {
+            Some(Callback::Inline(inline)) => SkipCallback::Inline(inline),
+            Some(Callback::Label(label)) => SkipCallback::Label(label),
+            Some(Callback::Skip(_)) => {
+                // Probably not reachable
+                return None;
```

Can you comment in this being unreachable? If we can guarantee this, we should probably use `unreachable!`, so we convey the right message. If an error occurs, it should be during compile time, and indicate that something went wrong (instead of silently returning `None`).

---

The function that it calls, self.parse_callback, can only return None or Some(Callback::Inline) or Some(Callback::Label). Should that be an `unreachable!` or would a `self.err` be more appropriate?

---

I've added a self.err, but let me know if I should change it.

---

If that should really be unreachable, then `unreachable!` is fine. If an usage error can lead to this, then `self.err` is more appropriate (because this is the user's mistake).